### ISSUE TITLE
Modify the security configuration steps to avoid securityadmin use

### ISSUE THREAD

**Issue by okynos** [wazuh/wazuh-packages#1968]
# Description
Hello!

We have detected in this issue https://github.com/wazuh/wazuh-packages/issues/1582 that there is a configuration option to set up automatically the security configuration without running security admin tool which is deprecated.

The parameter is `plugins.security.allow_default_init_securityindex: true`

# Tasks

- [ ] Include the parameter as a default in the package installation.
- [ ] Build and deploy the modified package.
- [ ] Test in all recommended systems.
 
# Validation

- [ ] The indexer and dashboard start and work without running securityadmin.
- [ ] There are less errors in the log as issue #1582 states.

**Comment by davidcr01** [comment]
# Update Report

## Testing

I will test the behavior of the indexer in multiple scenarios: the installation of a single-node indexer and a distributed installation.

In these tests, the `/etc/wazuh-indexer/opensearch.yml` is modified by adding the `plugins.security.allow_default_init_securityindex: true` option.

### Single node

**Installation and stating service**
<details><summary>Show log</summary>

```
root@ubuntu-focal:/home/vagrant# systemctl daemon-reload
root@ubuntu-focal:/home/vagrant# systemctl enable wazuh-indexer
Synchronizing state of wazuh-indexer.service with SysV service script with /lib/systemd/systemd-sysv-install.
Executing: /lib/systemd/systemd-sysv-install enable wazuh-indexer
Created symlink /etc/systemd/system/multi-user.target.wants/wazuh-indexer.service → /lib/systemd/system/wazuh-indexer.service.
root@ubuntu-focal:/home/vagrant# systemctl start wazuh-indexer
root@ubuntu-focal:/home/vagrant# systemctl sttus wazuh-indexer
Unknown operation sttus.
root@ubuntu-focal:/home/vagrant# systemctl status wazuh-indexer
● wazuh-indexer.service - Wazuh-indexer
     Loaded: loaded (/lib/systemd/system/wazuh-indexer.service; enabled; vendor preset: enabled)
     Active: active (running) since Tue 2023-02-21 10:28:33 UTC; 8s ago
       Docs: https://documentation.wazuh.com
   Main PID: 3403 (java)
      Tasks: 60 (limit: 3545)
     Memory: 1.2G
     CGroup: /system.slice/wazuh-indexer.service
             └─3403 /usr/share/wazuh-indexer/jdk/bin/java -Xshare:auto -Dopensearch.networkaddress.cache.ttl=60 -Dopensearch.networkaddress.cache.ne>

Feb 21 10:28:13 ubuntu-focal systemd[1]: Starting Wazuh-indexer...
Feb 21 10:28:20 ubuntu-focal systemd-entrypoint[3403]: WARNING: A terminally deprecated method in java.lang.System has been called
Feb 21 10:28:20 ubuntu-focal systemd-entrypoint[3403]: WARNING: System::setSecurityManager has been called by org.opensearch.bootstrap.OpenSearch (f>
Feb 21 10:28:20 ubuntu-focal systemd-entrypoint[3403]: WARNING: Please consider reporting this to the maintainers of org.opensearch.bootstrap.OpenSe>
Feb 21 10:28:20 ubuntu-focal systemd-entrypoint[3403]: WARNING: System::setSecurityManager will be removed in a future release
Feb 21 10:28:21 ubuntu-focal systemd-entrypoint[3403]: WARNING: A terminally deprecated method in java.lang.System has been called
Feb 21 10:28:21 ubuntu-focal systemd-entrypoint[3403]: WARNING: System::setSecurityManager has been called by org.opensearch.bootstrap.Security (fil>
Feb 21 10:28:21 ubuntu-focal systemd-entrypoint[3403]: WARNING: Please consider reporting this to the maintainers of org.opensearch.bootstrap.Securi>
```
</details>

**wazuh-cluster.log**
<details><summary>Show log</summary>

```
[2023-02-21T10:28:33,682][INFO ][o.o.c.r.a.AllocationService] [node-1] Cluster health status changed from [YELLOW] to [GREEN] (reason: [shards started [[.opendistro_security][0]]]).
[2023-02-21T10:28:33,708][INFO ][o.o.a.u.d.DestinationMigrationCoordinator] [node-1] Detected cluster change event for destination migration
[2023-02-21T10:28:33,711][INFO ][o.o.s.c.ConfigurationRepository] [node-1] Index .opendistro_security created?: true
[2023-02-21T10:28:33,712][INFO ][o.o.s.c.ConfigurationRepository] [node-1] Node started, try to initialize it. Wait for at least yellow cluster state....
[2023-02-21T10:28:33,717][INFO ][o.o.s.s.ConfigHelper     ] [node-1] Will update 'config' with /etc/wazuh-indexer/opensearch-security/config.yml and populate it with empty doc if file missing and populateEmptyI>
[2023-02-21T10:28:33,839][INFO ][o.o.c.m.MetadataMappingService] [node-1] [.opendistro_security/nvVCSqIiSvyTnZE6SQMPkQ] create_mapping
[2023-02-21T10:28:33,859][INFO ][o.o.a.u.d.DestinationMigrationCoordinator] [node-1] Detected cluster change event for destination migration
[2023-02-21T10:28:33,949][INFO ][o.o.s.s.ConfigHelper     ] [node-1] Doc with id 'config' and version 2 is updated in .opendistro_security index.
[2023-02-21T10:28:33,949][INFO ][o.o.s.s.ConfigHelper     ] [node-1] Will update 'roles' with /etc/wazuh-indexer/opensearch-security/roles.yml and populate it with empty doc if file missing and populateEmptyIfF>
[2023-02-21T10:28:33,978][INFO ][o.o.c.m.MetadataMappingService] [node-1] [.opendistro_security/nvVCSqIiSvyTnZE6SQMPkQ] update_mapping [_doc]
[2023-02-21T10:28:33,996][INFO ][o.o.a.u.d.DestinationMigrationCoordinator] [node-1] Detected cluster change event for destination migration
[2023-02-21T10:28:34,026][INFO ][o.o.s.s.ConfigHelper     ] [node-1] Doc with id 'roles' and version 2 is updated in .opendistro_security index.
[2023-02-21T10:28:34,027][INFO ][o.o.s.s.ConfigHelper     ] [node-1] Will update 'rolesmapping' with /etc/wazuh-indexer/opensearch-security/roles_mapping.yml and populate it with empty doc if file missing and p>
[2023-02-21T10:28:34,048][INFO ][o.o.c.m.MetadataMappingService] [node-1] [.opendistro_security/nvVCSqIiSvyTnZE6SQMPkQ] update_mapping [_doc]
[2023-02-21T10:28:34,068][INFO ][o.o.a.u.d.DestinationMigrationCoordinator] [node-1] Detected cluster change event for destination migration
[2023-02-21T10:28:34,092][INFO ][o.o.s.s.ConfigHelper     ] [node-1] Doc with id 'rolesmapping' and version 2 is updated in .opendistro_security index.
[2023-02-21T10:28:34,092][INFO ][o.o.s.s.ConfigHelper     ] [node-1] Will update 'internalusers' with /etc/wazuh-indexer/opensearch-security/internal_users.yml and populate it with empty doc if file missing and>
[2023-02-21T10:28:34,112][INFO ][o.o.c.m.MetadataMappingService] [node-1] [.opendistro_security/nvVCSqIiSvyTnZE6SQMPkQ] update_mapping [_doc]
[2023-02-21T10:28:34,137][INFO ][o.o.a.u.d.DestinationMigrationCoordinator] [node-1] Detected cluster change event for destination migration
[2023-02-21T10:28:34,160][INFO ][o.o.s.s.ConfigHelper     ] [node-1] Doc with id 'internalusers' and version 2 is updated in .opendistro_security index.
[2023-02-21T10:28:34,161][INFO ][o.o.s.s.ConfigHelper     ] [node-1] Will update 'actiongroups' with /etc/wazuh-indexer/opensearch-security/action_groups.yml and populate it with empty doc if file missing and p>
[2023-02-21T10:28:34,179][INFO ][o.o.c.m.MetadataMappingService] [node-1] [.opendistro_security/nvVCSqIiSvyTnZE6SQMPkQ] update_mapping [_doc]
[2023-02-21T10:28:34,199][INFO ][o.o.a.u.d.DestinationMigrationCoordinator] [node-1] Detected cluster change event for destination migration
[2023-02-21T10:28:34,215][INFO ][o.o.s.s.ConfigHelper     ] [node-1] Doc with id 'actiongroups' and version 2 is updated in .opendistro_security index.
[2023-02-21T10:28:34,216][INFO ][o.o.s.s.ConfigHelper     ] [node-1] Will update 'tenants' with /etc/wazuh-indexer/opensearch-security/tenants.yml and populate it with empty doc if file missing and populateEmpt>
[2023-02-21T10:28:34,237][INFO ][o.o.c.m.MetadataMappingService] [node-1] [.opendistro_security/nvVCSqIiSvyTnZE6SQMPkQ] update_mapping [_doc]
[2023-02-21T10:28:34,253][INFO ][o.o.a.u.d.DestinationMigrationCoordinator] [node-1] Detected cluster change event for destination migration
[2023-02-21T10:28:34,273][INFO ][o.o.s.s.ConfigHelper     ] [node-1] Doc with id 'tenants' and version 2 is updated in .opendistro_security index.
[2023-02-21T10:28:34,273][INFO ][o.o.s.s.ConfigHelper     ] [node-1] Will update 'nodesdn' with /etc/wazuh-indexer/opensearch-security/nodes_dn.yml and populate it with empty doc if file missing and populateEmp>
[2023-02-21T10:28:34,296][INFO ][o.o.c.m.MetadataMappingService] [node-1] [.opendistro_security/nvVCSqIiSvyTnZE6SQMPkQ] update_mapping [_doc]
[2023-02-21T10:28:34,319][INFO ][o.o.a.u.d.DestinationMigrationCoordinator] [node-1] Detected cluster change event for destination migration
[2023-02-21T10:28:34,340][INFO ][o.o.s.s.ConfigHelper     ] [node-1] Doc with id 'nodesdn' and version 2 is updated in .opendistro_security index.
[2023-02-21T10:28:34,341][INFO ][o.o.s.s.ConfigHelper     ] [node-1] Will update 'whitelist' with /etc/wazuh-indexer/opensearch-security/whitelist.yml and populate it with empty doc if file missing and populate>
[2023-02-21T10:28:34,361][INFO ][o.o.c.m.MetadataMappingService] [node-1] [.opendistro_security/nvVCSqIiSvyTnZE6SQMPkQ] update_mapping [_doc]
[2023-02-21T10:28:34,381][INFO ][o.o.a.u.d.DestinationMigrationCoordinator] [node-1] Detected cluster change event for destination migration
[2023-02-21T10:28:34,404][INFO ][o.o.s.s.ConfigHelper     ] [node-1] Doc with id 'whitelist' and version 2 is updated in .opendistro_security index.
[2023-02-21T10:28:34,404][INFO ][o.o.s.s.ConfigHelper     ] [node-1] Will update 'allowlist' with /etc/wazuh-indexer/opensearch-security/allowlist.yml and populate it with empty doc if file missing and populate>
[2023-02-21T10:28:34,433][INFO ][o.o.c.m.MetadataMappingService] [node-1] [.opendistro_security/nvVCSqIiSvyTnZE6SQMPkQ] update_mapping [_doc]
[2023-02-21T10:28:34,453][INFO ][o.o.a.u.d.DestinationMigrationCoordinator] [node-1] Detected cluster change event for destination migration
[2023-02-21T10:28:34,472][INFO ][o.o.s.s.ConfigHelper     ] [node-1] Doc with id 'allowlist' and version 2 is updated in .opendistro_security index.
[2023-02-21T10:28:34,472][INFO ][o.o.s.s.ConfigHelper     ] [node-1] Will update 'audit' with /etc/wazuh-indexer/opensearch-security/audit.yml and populate it with empty doc if file missing and populateEmptyIfF>
[2023-02-21T10:28:34,520][INFO ][o.o.c.m.MetadataMappingService] [node-1] [.opendistro_security/nvVCSqIiSvyTnZE6SQMPkQ] update_mapping [_doc]
[2023-02-21T10:28:34,540][INFO ][o.o.a.u.d.DestinationMigrationCoordinator] [node-1] Detected cluster change event for destination migration
[2023-02-21T10:28:34,557][INFO ][o.o.s.s.ConfigHelper     ] [node-1] Doc with id 'audit' and version 2 is updated in .opendistro_security index.
[2023-02-21T10:28:34,720][INFO ][o.o.s.a.i.AuditLogImpl   ] [node-1] Auditing on REST API is enabled.
```
</details>

**Found errors**
- The first message is not an error itself.
- The second is a general ERROR message during the node startup.

:heavy_check_mark: The number of errors are reduced.
<details><summary>Show log</summary>

```
root@ubuntu-focal:/home/vagrant# cat /var/log/wazuh-indexer/wazuh-cluster.log | grep -i -e "Error"
[2023-02-21T10:28:21,795][INFO ][o.o.n.Node               ] [node-1] JVM arguments [-Xshare:auto, -Dopensearch.networkaddress.cache.ttl=60, -Dopensearch.networkaddress.cache.negative.ttl=10, -XX:+AlwaysPreTouch, -Xss1m, -Djava.awt.headless=true, -Dfile.encoding=UTF-8, -Djna.nosys=true, -XX:-OmitStackTraceInFastThrow, -XX:+ShowCodeDetailsInExceptionMessages, -Dio.netty.noUnsafe=true, -Dio.netty.noKeySetOptimization=true, -Dio.netty.recycler.maxCapacityPerThread=0, -Dio.netty.allocator.numDirectArenas=0, -Dlog4j.shutdownHookEnabled=false, -Dlog4j2.disable.jmx=true, -Djava.locale.providers=SPI,COMPAT, -Xms1g, -Xmx1g, -XX:+UseG1GC, -XX:G1ReservePercent=25, -XX:InitiatingHeapOccupancyPercent=30, -Djava.io.tmpdir=/tmp/opensearch-3285845650814918236, -XX:+HeapDumpOnOutOfMemoryError, -XX:HeapDumpPath=/var/lib/wazuh-indexer, -XX:ErrorFile=/var/log/wazuh-indexer/hs_err_pid%p.log, -Xlog:gc*,gc+age=trace,safepoint:file=/var/log/wazuh-indexer/gc.log:utctime,pid,tags:filecount=32,filesize=64m, -Dclk.tck=100, -Djdk.attach.allowAttachSelf=true, -Djava.security.policy=file:///etc/wazuh-indexer/opensearch-performance-analyzer/opensearch_security.policy, --add-opens=jdk.attach/sun.tools.attach=ALL-UNNAMED, -Dclk.tck=100, -Djdk.attach.allowAttachSelf=true, -Djava.security.policy=file:///usr/share/wazuh-indexer/plugins/opendistro-performance-analyzer/pa_config/es_security.policy, -XX:MaxDirectMemorySize=536870912, -Dopensearch.path.home=/usr/share/wazuh-indexer, -Dopensearch.path.conf=/etc/wazuh-indexer, -Dopensearch.distribution.type=rpm, -Dopensearch.bundled_jdk=true]
[2023-02-21T10:28:29,931][ERROR][o.o.s.a.s.SinkProvider   ] [node-1] Default endpoint could not be created, auditlog will not work properly.
```
</details>

**Comment by davidcr01** [comment]
# Update Report

## Development

The indexer packages were generated by running the `generate_package.sh` of the `stack/indexer/deb` and `stack/indexer/rpm` folders. 

If we install these new packages, we can see that the new option is added in the `opensearch.yml` configuration file:

```
root@ubuntu-focal:/home/vagrant# cat /etc/wazuh-indexer/opensearch.yml | grep default_init_securityindex

plugins.security.allow_default_init_securityindex: true
``` 


:warning: This issue goes to review until its approval by the team.

**Comment by davidcr01** [comment]
# Update Report

## Test

A test has been done to check the behavior of this option. I tried to:
- Install the Wazuh indexer without the new parameter.
- Change the passwords
- Make a backup of the `internal_users.yml` file.
- Reinstall the Wazuh indexer with the new parameter.
- Remove the `internal_users.yml` default file and copy the `internal_users.yml` modified file.

This test checks if the Wazuh indexer with the new parameter (run securityadmin by default) loads the modified `internal_users.yml`.

:red_circle: The result is that the Wazuh indexer reads this configuration but it does not use it. Instead, it uses the default configuration. If we read the description of the parameter:

> When set to true, the security plugin uses default security settings if an attempt to create the security index fails when OpenSearch launches

In conclusion, if we want to change the passwords of the `internal_users.yml` file, the `securityadmin` plugin has to be executed to load the modified security index, and that is actually what we want to avoid. This parameter seems not to solve the proposed problem.

**Comment by davidcr01** [comment]
This issue goes to "On hold" until OpenSearch releases the `securityadmin` alternative. https://github.com/opensearch-project/security/issues/1755

**Comment by teddytpc1** [comment]
Blocked by: https://github.com/opensearch-project/security/issues/1755
